src/linearshrinkage.jl

```diff
@@ -296,26 +312,27 @@ Compute the shrinkage estimator where the target is a `DiagonalCommonVariance`.
 """
 function linear_shrinkage(::DiagonalCommonVariance, Xc::AbstractMatrix,
                           S::AbstractMatrix, λ::Symbol, n::Int, p::Int,
-                          corrected::Bool)
+                          corrected::Bool, weights::FrequencyWeights...)
 
     F   = target_B(S, p)
     T   = float(eltype(F))
-    κ   = n - Int(corrected)
-    γ   = T(κ/n)
+    wn = totalweight(n, weights...)
+    κ   = wn - Int(corrected)
+    γ   = T(κ/wn)
     Xc² = Xc.^2
     # computing the shrinkage
     if λ ∈ [:auto, :lw]
         v   = F.λ # tr(S)/p
         ΣS² = sumij2(S, with_diag=true)
-        λ   = sumij(uccov(Xc²), with_diag=true) / γ^2 - ΣS²
+        λ   = sumij(uccov(Xc², weights...), with_diag=true) / γ^2 - ΣS²
         λ  /= κ * (ΣS² - p*v^2)
     elseif λ == :ss
         # use the standardised data matrix
-        d   = one(T) ./ vec(sum(Xc², dims=1))
+        d   = one(T) ./ vec(sum(Xc², weights...; dims=1))
         S̄   = rescale(S, sqrt.(d)) # this has diagonal 1/κ
         v̄   = κ # tr(S̄)/p
         ΣS̄² = sumij2(S̄, with_diag=true)
-        λ   = sumij(rescale!(uccov(Xc²), d), with_diag=true) / γ^2 - ΣS̄²
+        λ   = sumij(rescale!(uccov(Xc², weights...), d), with_diag=true) / γ^2 - ΣS̄²
         λ  /= T(κ * ΣS̄² - p/κ)
     elseif λ == :rblw
```

Comment:
We probably also want to use `wn` instead of `n` for `λ` calculation in this method.